### PR TITLE
Add support for global_startindex != 0 to covering_grid

### PR DESCRIPTION
**Pull request by atmyers**
Fixes Issue #3467 

Script to reproduce problem:

```python
# Import statements
import yt
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

diag_name = 'diag3' # E.g., diagnostics.diags_names = diag
iteration = 1
plotfile = './diags/{}{:05d}'.format(diag_name, iteration)
field = 'rho'
ds = yt.load( plotfile ) # Load the plotfile
# ds.field_list # Print all available quantities

# Get field quantities
all_data_level_0 = ds.covering_grid(level=0,left_edge=ds.domain_left_edge, dims=ds.domain_dimensions)
Bx = all_data_level_0['boxlib', field].v.squeeze()

extent = [ds.domain_left_edge[ds.dimensionality-1], ds.domain_right_edge[ds.dimensionality-1],
              ds.domain_left_edge[0], ds.domain_right_edge[0] ]

# Plot image
fig, ax = plt.subplots()
clim = 1e-19
ax.imshow(Bx,aspect='auto', cmap = 'seismic', clim = [-clim, clim], extent= extent, origin='lower')
```

Dataset is attached.
[diag300001.tar.gz](https://github.com/yt-project/yt/files/6957948/diag300001.tar.gz)

### CODE REVIEW

**Review by neutrinoceros — CHANGES_REQUESTED** [review]
Here's a couple questions I'd like to resolve before merging this. Note that style check failed,
see https://yt-project.org/doc/developing/developing.html#automatically-checking-and-fixing-code-style

File: yt/data_objects/construction_data_containers.py
Comment (atmyers) [reply yt/data_objects/construction_data_containers.py]: The getter is assumed to be present already, on line 720 of `construction_data_containers.py`:
```python
    @property
    def icoords(self):
        ic = np.indices(self.ActiveDimensions).astype("int64")
        return np.column_stack(
            [i.ravel() + gi for i, gi in zip(ic, self.get_global_startindex())]
        )
```

So I guess this is a bug on `main`. Fine with me if you prefer to access `global_startindex` directly.

Comment (neutrinoceros) [reply yt/data_objects/construction_data_containers.py]: I see. Indeed the `YTCoveringGrid` class' current implementation seems to be broken. I also checked that the missing accessor isn't implemented either in child classes, so it doesn't seem to be a case of an abstract class that's not explicitly declared as such. I think this means that the `icoords` property is in fact never accessed in any of our tests. I would defer to @matthewturk on this. Your patch seems acceptable to me, but I'm not sure wether we need to keep this property.

File: yt/data_objects/static_output.py
Comment (neutrinoceros) [inline yt/data_objects/static_output.py:140]:
```suggestion
    # the point in index space "domain_left_edge" doesn't necessarily 
    # map to (0, 0, 0)
```

**Comment by neutrinoceros** [comment]
For posterity here's a look at the result from the embedded script, before and after the patch

before
before3468

after
after3468

**Review by neutrinoceros — APPROVED** [review]
Besides my last small suggestion on the comment phrasing, I'm now happy with the state of the code, and I checked that it indeed fixes the issue addressed.

**Pull request merged**
